gui/src/main/tray.window.js

```diff
@@ -12,114 +12,100 @@ const log = require('cozy-desktop').default.logger({
   component: 'GUI'
 })
 
-const behaviours = require('./window_behaviour')
+const WindowManager = require('./window_behaviour')
 
-let win = null
-let app = null
-let desktop = null
-
-module.exports.init = (appRef, desktopRef) => {
-  desktop = desktopRef
-  app = appRef
-}
-
-module.exports.create = () => {
-  log.debug({'create': 'onboarding'})
-  if (win != null) return
-  win = new BrowserWindow({
-    title: 'TRAY',
-    icon: `${__dirname}/images/icon.png`,
-    windowPosition: (process.platform === 'win32') ? 'trayBottomCenter' : 'trayCenter',
-    frame: false,
-    show: false,
-    skipTaskbar: true,
-    // transparent: true,
-    width: DASHBOARD_SCREEN_WIDTH,
-    height: DASHBOARD_SCREEN_HEIGHT
-  })
-  win.loadURL(`file://${__dirname}/../../index.html#dashboard`)
-  behaviours.noMenu(win)
-  // behaviours.devTools(win)
-  behaviours.dockApple(app, win)
-  behaviours.openExternalLinks(win)
-  win.on('closed', () => { win = null })
-  win.on('blur', () => setTimeout(() => { if (!win.isFocused() && !win.isDevToolsFocused()) win.close() }, 400))
-  win.positioner = new Positioner(win)
-}
-
-module.exports.onReady = (cb) => {
-  win && win.webContents && win.webContents.once('dom-ready', () => {
-    setTimeout(cb, 100) // elm initialization
-  })
-}
-
-module.exports.send = (...args) => {
-  if (win && win.webContents) {
-    win.webContents.send(...args)
+module.exports = class TrayWM extends WindowManager {
+  windowOptions () {
+    return {
+      title: 'TRAY',
+      windowPosition: (process.platform === 'win32') ? 'trayBottomCenter' : 'trayCenter',
+      frame: false,
+      show: false,
+      skipTaskbar: true,
+      // transparent: true,
+      width: DASHBOARD_SCREEN_WIDTH,
+      height: DASHBOARD_SCREEN_HEIGHT
+    }
   }
-}
-
-module.exports.reload = () => {
-  if (win) win.reload()
-}
 
-module.exports.show = (trayPos) => {
-  if (!win) module.exports.create()
-  let pos = null
-  if (trayPos === undefined || trayPos.x === 0) {
-    pos = (process.platform === 'win32') ? 'bottomRight' : 'topRight'
-  } else {
-    pos = (process.platform === 'win32') ? 'trayBottomCenter' : 'trayCenter'
+  create () {
+    log.debug({tcreate: true})
+    let pReady = super.create()
```

Comment:
Is this some kind of prefixing convention for js promises?

Reply:
Not really a convention, just an habit I picked when dealing with mixed promise/cb codebases